src/VssDatabase.cpp

```diff
@@ -42,6 +42,53 @@ VssDatabase::VssDatabase(std::shared_ptr<ILogger> loggerUtil,
 
 VssDatabase::~VssDatabase() {}
 
+/** Returns true if the json structure is of VSS type sensor */
+bool VssDatabase::isActor(const json &element) {
+  return element.contains("type") && element["type"]=="actuator";
+}
+
+/** Returns true if the json structure is of VSS type sensor */
+bool VssDatabase::isSensor(const json &element) {
+  return element.contains("type") && element["type"]=="sensor";
+}
+
+/** Returns true if the json structure is of VSS type attribute */
+bool VssDatabase::isAttribute(const json &element) {
+  return element.contains("type") && element["type"]=="attribute";
+}
+
+
+/** Iterates over a given VSS tree and checks for sensors and actuators specifying the "default"
+ *  metadata. If a default is present, it will be used as "value" (and thus returned upon get
+ *  even if there has been no previous set)
+ */ 
+void VssDatabase::applyDefaultValues(json &tree, VSSPath path) {
+  //logger_->Log(LogLevel::VERBOSE, "Applying default values in "+path.to_string());
+
+  if ( isSensor(tree) || isActor(tree) || isAttribute(tree)) {
```

Comment:
nothing. Only Sensors/Actors and Attributes can have default values. Strictly speaking, currently something that is not not sensor/actor/tree/default is a VSS spec violation, similarly a branch with a default should never happen. But VSS compliance of the whole tree should be checked by the VSS tooling, or  - if in KUKSA -  in another step.

You are right default seems to be only specified for attributes https://genivi.github.io/vehicle_signal_specification/rule_set/data_entry/attributes/  , it is however used for other elements to, see https://github.com/GENIVI/vehicle_signal_specification/blob/9c82e7018b05958706107bf86a7e30feca970c2b/spec/Powertrain/Battery.vspec#L111   

Will raise an issue

Reply:
For reference https://github.com/GENIVI/vehicle_signal_specification/issues/283
For now I would go with the spec (i.e. also process it for sensors/actuators), and change the code if VSS decides otherwise